### PR TITLE
QnAMaker DTO changes for Multiturn

### PR DESCRIPTION
**Pull request by ghost**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5909 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5909/commits/f607d8d3142da75878049c63e9929c828005760f

**Comment by dsgouda** [comment]
@praries880 Taking over

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 1]
```diff
@@ -671,6 +679,32 @@
         }
       }
     },
+    "UpdateContextDTO": {
+      "type": "object",
+      "description": "Update Body schema to represent context to be updated",
+      "additionalProperties": false,
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: Remove this line

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 2]
```diff
@@ -671,6 +679,32 @@
...
+      "properties": {
+        "promptsToDelete": {
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: Please mark the readonly properties where applicable

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 3]
```diff
@@ -870,6 +904,63 @@
...
+          "items": {
+            "$ref": "#/definitions/PromptDTO"
+          }
+        }
+      }
+    },
+    "PromptDTO": {
+      "type": "object",
+      "description": "Prompt for an answer.",
+      "additionalProperties": false,
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: Remove this line

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 4]
```diff
@@ -870,6 +904,63 @@
...
+      "properties": {
+        "isContextOnly": {
+          "type": "boolean",
+          "description": "To mark if a prompt is relevant only with a previous question or not.\ntrue - Do not include this QnA as search result for queries without context\nfalse - ignores context and includes this QnA in search result"
+        },
+        "prompts": {
+          "type": "array",
+          "description": "List of prompts associated with the answer.",
+          "maxLength": 20,
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: Use `"maxItems": 20` instead

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 5]
```diff
@@ -671,6 +679,32 @@
...
+      "properties": {
+        "promptsToDelete": {
+          "type": "array",
+          "description": "List of prompts associated with qna to be deleted",
+          "items": {
+            "type": "integer",
+            "format": "int32"
+          }
+        },
+        "promptsToAdd": {
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: It looks like some properties are shared between `UpdateContextDTO` and `ContextDTO`

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json (hunk 5 above)
Comment (dsgouda) [reply specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: Is this resolved? I don't see any corresponding changes

Comment (bingisbestest) [reply specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: I added comment not sure why it got deleted. There is only one bool property in common rest all are different.

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json [hunk 6]
```diff
@@ -870,6 +903,62 @@
...
+          "items": {
+            "$ref": "#/definitions/PromptDTO"
+          }
+        }
+      }
+    },
+    "PromptDTO": {
+      "type": "object",
+      "description": "Prompt for an answer.",
+      "additionalProperties": false,
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: Not sure if this line was removed

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json (hunk 2 above)
Comment (bingisbestest) [reply specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: These are all read/write

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@sarangan12 Could you please take a look at the Ruby SDK failures?

**Review by dsgouda — APPROVED** [review]
LGTM will merge once CIs pass

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json (hunk 6 above)
Comment (bingisbestest) [reply specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json]: missed it. removed.

**Pull request merged**